app/components/windows/AddSource.vue.ts

```diff
@@ -44,7 +44,7 @@ export default class AddSource extends Vue {
     return { name: source.name, value: source.sourceId };
   });
 
-  selectedSourceId = this.sources[0].sourceId;
+  selectedSourceId = this.sources[0] ? this.sources[0].sourceId : null;
```

Comment:
does this mean `sources` is either an array or a single source? i think we should standardize our data shapes here to always be arrays if that's the case

Reply:
PS this is the *one* thing I miss from CoffeeScript.  In CoffeeScript you can use the existential operator to skip null checks:
```CoffeeScript
selectedSourceId = this.sources[0]?.sourceId;
```

Reply:
there's a proposal for safe navigation in ES; i think it's going to be added somewhat soon

Reply:
oh nice